app/services/ClientManagerService.js

```diff
@@ -1,18 +1,40 @@
-function ClientManagerService(){
-  var clients = [];
+var ClientFactory = require('../factories/ClientFactory');
 
-  return {
-    connected: connected,
-    disconnected: disconnected,
-  };
+function ClientManagerService() {
+    var clients = [];
 
-  function connected(client){
-    clients.push(client);
-  }
+    return {
+        connected: connected,
+        disconnected: disconnected,
+        setData: setData,
+        create: create,
+        getClients: getClients
+    };
 
-  function disconnected(client){
-    clients.splice(clients.indexOf(client), 1);
-  }
+    function connected(client) {
+        clients.push(client);
+    }
+
+    function setData(client, data) {
```

Comment:
`setData` is too generic, something like `setUserProfile` or whatever would be better